### ISSUE TITLE
E2EE release for android

### ISSUE THREAD

**Issue by tobiasKaminsky** [nextcloud/android#1788]
@mannp
> Hi all, I have updated to the latest server code and am wondering where I might find any updated E2EE release for android (if you've had chance to create any yet)?

> I am in the nextcloud beta + alpha google+ group on the play store but that doesn't seem to have the option to encrypt a directory?

> Thanks in advance

**Comment by tobiasKaminsky** [comment]
It should be sufficient to be in the alpha google+ group. Then you should automatically get an update via play store.
What version is shown in the app -> settings -> Nextcloud Android app?

**Comment by mannp** [comment]
@tobiasKaminsky it shows 2.0.1 and beta in in the play store.

When I went to the alpha group it said 'member'.

**Comment by cleary** [comment]
I'm having the same issue.
If I access my NextCloud from a browser, there's a header bar indicating that end to end encryption (alpha) is available, but my Android client does not show the menu option for encrypting empty folders.

On my phone:
Nextcloud (Beta) is installed, v2.0.0
Phone: Samsung S8
Android: 7.0
Location: Australia
I'm a member of the Google+ group as well...

**Comment by tobiasKaminsky** [comment]
We will soon release a new E2E enabled version on alpha channel.

**Comment by cleary** [comment]
@tobiasKaminsky thanks, but I don't seem to have access to the alpha channel (despite following the instructions). What else can I do to get access?

**Comment by tobiasKaminsky** [comment]
All you have to do is to be in the google+ group.

**Comment by mannp** [comment]
Mmh, with 2 people saying they are clearly in the group and don't get the alpha update offered, is there something else we need to check?

**Comment by e-alfred** [comment]
The dev version on F-Droid is pretty outdated (last release at the end of September), can you make a new release with encryption support there as well?

https://f-droid.org/packages/com.nextcloud.android.beta/

**Comment by tobiasKaminsky** [comment]
The e2e alpha version was outdated by google the moment we released 2.0.
As I am working currently on an improvement of E2E there will be a new E2E release the next time

@e-alfred the dev version will soon reflect the latest master version, so E2E will as soon in it as we merge it.

**Comment by mannp** [comment]
@tobiasKaminsky so where should we be looking for your recently merged 19 updates.

Will that come through the alpha channel / as a beta update or a general update?

**Comment by tobiasKaminsky** [comment]
This pull request https://github.com/nextcloud/android/pull/1598 with latest master will be version 2.1.0-e2e-01, available on alpha channel on gplay.
Beta channel on gplay will be our 2.1.0 rc1-x which helps us to stabilize the upcoming release (currently no estimation when this will be).
Completely independent from that we have our dev version on fdroid or as manually download. This will (soon) include always lastest master and can be installed in parallel.

**Comment by cleary** [comment]
@tobiasKaminsky Thanks! That makes sense now :)

**Comment by mannp** [comment]
Hi @tobiasKaminsky I see some increasing commits for the android client and am keen to try :)

**Comment by tobiasKaminsky** [comment]
We recently changed some internal behaviour and I want to make sure that it is still compatible with iOS. After that I will release a new version.

**Comment by tobiasKaminsky** [comment]
@mannp a new E2E alpha version is now released and should be available soon.

**Comment by mannp** [comment]
Hi @tobiasKaminsky excellent, thanks. Just had a google+ notification telling me the same, so that works :)

No update on google play as yet, but keep an eye out through the day.

**Comment by tobiasKaminsky** [comment]
@mannp on my test device I received an update.
You can verify it in our app: settings -> Nextcloud Android app: 2.1.0-e2e-01

**Comment by mannp** [comment]
@tobiasKaminsky Nothing here, just checked a couple of minutes ago. Still on 2.01.

**Comment by cleary** [comment]
Thanks @tobiasKaminsky, I'm still not seeing it either though :/

**Comment by mannp** [comment]
@tobiasKaminsky I wondered if you have built this with a restricted device profile as I have received it on my Xperia ultra but not on two newer 64bit devices...? All using the same account in beta and alpha channel...

**Comment by cleary** [comment]
@mannp @tobiasKaminsky I setup a new device this morning (Samsung Galaxy Tab S3) - brand new, high end tablet (ref @mannp 64bit comment) - 
I installed Nextcloud (beta), and didn't notice anything different as far as Play Store or installation went.
Logged in and the feature is available!

So I uninstalled from my Samsung S8, reinstalled and voila! We have encryption :)

Hopefully an uninstall and reinstall is all that's required to force it -

**Comment by mannp** [comment]
Thanks @cleary reinstall worked for me, just a pity e2ee is now nc13 only.

Now just need a nc13 instance to test :)

**Comment by cleary** [comment]
@mannp Heh, immediately after posting I went further than looking for the item in the menu... And couldn't get it to function.
I'll have some time tomorrow to upgrade my instance (I'm just tracking the git report for testing atm)

**Comment by mannp** [comment]
Hi @cleary it didn't work for me either, with this latest android update, but the latest git has made the plugin only for nc13, so I wasn't sure if that was the reason why it failed to create the keys.

I don't have a nc13 instance so I won't be testing in the short term.

**Comment by e-alfred** [comment]
I downloaded the latest Nextcloud dev 20171204 version from F-Droid but I can't enable encryption with Nextcloud 13 Beta 3 and the end-to-end encryption endpoint. Is E2EE included in the latest dev app on F-Droid?

**Comment by AndyScherzinger** [comment]
E2e is only available as alpha release via Google Play Store but not part of the dev builds yet.

**Comment by cleary** [comment]
@mannp still getting the same error on a NC13 instance (v13.0.0beta3 git tag)
Migration completed, everything else is working fine, but I'm still getting the error on the android client (v2.1.0-e2e-01):
"Generating New Keys..."
"Error: Retrieving keys was unsuccessful, please try it again"

No amount of trying again is helping :(

**Comment by mannp** [comment]
@cleary thanks for the feedback. That was the same error I was getting and didn't know it could only be done by the admin account either.

@tobiasKaminsky is that by design, that normal users cannot create keys or does the admin account service other users, which is not what I would expect..

**Comment by cleary** [comment]
@mannp I logged a bug regardless: #1859

The only user controllable option was whether they were in the admin group or not, and that didn't make a difference, so it seemed bug-worthy to me :)

**Comment by GitHubUser4234** [comment]
Hi @tobiasKaminsky,

Just wanted to try out e2e, today I joined the Google+ group on my mobile (>Android 4.4) and installed the beta from Google Play for the first time. However the version is only 2.0 and no update is available to get it to alpha. Re-installation didn't solve the problem. Maybe something got outdated at Google again? 😸 

In that regard, it would certainly also help to finally merge e2e with the dev builds, thanks

**Comment by tobiasKaminsky** [comment]
It should work for non-admins. In fact I never use an admin user to test/delevop, so it is working fine for me all the time.
Unfortunately I currently have no idea what the problem could be, @schiessle?

E2E will be in dev builds once it is reviewed and merged. Currently there is one thing open and then the review process can start.

**Comment by GitHubUser4234** [comment]
Thanks @tobiasKaminsky for the reply about the dev builds status.

Strangely, it turned out that when I checked my mobile today, an update was finally available through Google Play and an update to alpha was successful, so had to wait about a day...

**Comment by tobiasKaminsky** [comment]
Unfortunately this is nothing we can influence, but glad that you received the update :-(
